File: 11기 코딩테스트_230308/연습2/MostAppearLetter/MostAppearLetter.py
def solution(s):
    """
    :param s: string
    :return: string
    """

    char_counter = {}
    for i in s:
        if i in char_counter:
            char_counter[i] += 1
        else:
            char_counter[i] = 1

    most_appear_char = None
    appear_count = 0
    for num, count in sorted(char_counter.items()):
        if appear_count < count:
            most_appear_char = num
            appear_count = count

    return most_appear_char

File: 11기 코딩테스트_230308/연습2/MostAppearLetter/MostAppearLetter.js
/**
 * @param s {string}
 * @returns {string}
 */
function solution(s) {
  const charCounter = []

  for (let i = 0; i < s.length; i++) {
    const c = s[i]
    const charIndex = charToIndex(c)
    charCounter[charIndex] = charCounter[charIndex] + 1 || 1
  }

  let mostAppearLetter = ''
  let appearCount = 0
  charCounter.forEach((count, charIndex) => {
    if (appearCount < count) {
      mostAppearLetter = indexToChar(charIndex)
      appearCount = count
    }
  })

  return mostAppearLetter
}

function charToIndex(c) {
  const aCode = 97
  return c.charCodeAt(0) - aCode
}

function indexToChar(i) {
  const aCode = 97
  return String.fromCharCode(i + aCode)
}

export default solution